packages/@uppy/drop-target/src/index.js

```diff
@@ -25,18 +27,24 @@ module.exports = class DropTarget extends Plugin {
     this.removeDragOverClassTimeout = null
   }
 
-  addFiles = (files) => {
-    const descriptors = files.map((file) => ({
-      source: this.id,
-      name: file.name,
-      type: file.type,
-      data: file,
-      meta: {
+  addFiles = (files, event) => {
+    const descriptors = files.map((file) => {
+      const meta = {
         // path of the file relative to the ancestor directory the user selected.
         // e.g. 'docs/Old Prague/airbnb.pdf'
         relativePath: file.relativePath || null,
-      },
-    }))
+      }
+      if (this.setMeta instanceof Function) {
+        Object.assign(meta, this.setMeta(file, event))
+      }
```

Comment:
If `setMeta` is neither nullish nor a function, it's likely a mistake IMHO throwing would be the right behavior. I think we can simplify this with optional chaining:
```suggestion
        ...this.setMeta?.(file, event),
      }
```